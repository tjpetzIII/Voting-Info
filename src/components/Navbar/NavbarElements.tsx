import type { ReactNode } from "react";
import { FaBars } from "react-icons/fa";
import { NavLink as RouterLink } from "react-router-dom";
import { clsx } from "clsx";
import React from "react";

interface NavProps {
  children: ReactNode;
}

interface NavLinkProps {
  to: string;
  children: ReactNode;
}
interface NavMenuProps {
  children: ReactNode;
}
// Nav wrapper
export const Nav = ({ children }: NavProps) => (
  <nav className="bg-blue-main h-20 flex justify-center mx-auto px-4">
    {children}
  </nav>
);

// NavLink
export const NavLink = ({ to, children }: NavLinkProps) => (
  <RouterLink
    to={to}
    className={({ isActive }) =>
      clsx(
        "font-mono text-xl flex items-center px-4 h-full cursor-pointer transition-all duration-200",
        {
          "underline underline-offset-4 text-sky-400": isActive,
          "text-white no-underline": !isActive,
        }
      )
    }
  >
    {children}
  </RouterLink>
);

// Hamburger icon
export const Bars = () => (
  <FaBars className="hidden text-gray-500 md:hidden block absolute top-0 right-0 transform -translate-x-full translate-y-3/4 text-[1.8rem] cursor-pointer" />
);

// Menu wrapper
export const NavMenu = ({ children }: NavMenuProps) => {
  const childrenArray = React.Children.toArray(children);
  const [first, ...rest] = childrenArray;

  return (
    <div className="w-full flex items-center justify-between px-4">
      {/* Left-aligned first item (e.g. icon) */}
      <div className="flex items-center">{first}</div>

      {/* Centered remaining nav items */}
      <div className="flex gap-4 justify-center flex-1">{rest}</div>
    </div>
  );
};
