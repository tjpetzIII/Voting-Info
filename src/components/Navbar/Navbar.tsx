import { Nav, NavLink, NavMenu } from "./NavbarElements";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPersonBooth } from "@fortawesome/free-solid-svg-icons";
const Navbar = () => {
  return (
    <>
      <Nav>
        <NavMenu>
          <FontAwesomeIcon
            icon={faPersonBooth}
            className="text-white-main text-4xl"
          />
          <NavLink to="/voting-info">Voting Info</NavLink>
          <NavLink to="/about">About</NavLink>
          <NavLink to="/contact">Contact Us</NavLink>
          <NavLink to="/sign-up">Sign Up</NavLink>
        </NavMenu>
      </Nav>
    </>
  );
};

export default Navbar;
