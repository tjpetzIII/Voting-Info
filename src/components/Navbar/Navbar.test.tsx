import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "./Navbar";

describe("Navbar", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
