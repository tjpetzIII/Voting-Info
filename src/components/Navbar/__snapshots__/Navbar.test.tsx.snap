// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-blue-main h-20 flex justify-center mx-auto px-4"
  >
    <div
      class="w-full flex items-center justify-between px-4"
    >
      <div
        class="flex items-center"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-person-booth text-white-main text-4xl"
          data-icon="person-booth"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 160 64 0 0-160zm320 0c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-448zM224 512c17.7 0 32-14.3 32-32l0-160-64 0 0 160c0 17.7 14.3 32 32 32zM320 0c-9.3 0-18.1 4-24.2 11s-8.8 16.3-7.5 25.5l31.2 218.6L288.6 409.7c-3.5 17.3 7.8 34.2 25.1 37.7s34.2-7.8 37.7-25.1l.7-3.6c1.3 16.4 15.1 29.4 31.9 29.4c17.7 0 32-14.3 32-32c0 17.7 14.3 32 32 32s32-14.3 32-32l0-384c0-17.7-14.3-32-32-32L320 0zM112 80A48 48 0 1 0 16 80a48 48 0 1 0 96 0zm0 261.3l0-72.1 4.7 4.7c9 9 21.2 14.1 33.9 14.1l73.4 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-66.7 0-41.6-41.6c-14.3-14.3-33.8-22.4-54-22.4C27.6 160 0 187.6 0 221.6l0 55.7 0 .9L0 480c0 17.7 14.3 32 32 32s32-14.3 32-32l0-96 32 42.7L96 480c0 17.7 14.3 32 32 32s32-14.3 32-32l0-58.7c0-10.4-3.4-20.5-9.6-28.8L112 341.3z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="flex gap-4 justify-center flex-1"
      >
        <a
          class="font-mono text-xl flex items-center px-4 h-full cursor-pointer transition-all duration-200 text-white no-underline"
          data-discover="true"
          href="/voting-info"
        >
          Voting Info
        </a>
        <a
          class="font-mono text-xl flex items-center px-4 h-full cursor-pointer transition-all duration-200 text-white no-underline"
          data-discover="true"
          href="/about"
        >
          About
        </a>
        <a
          class="font-mono text-xl flex items-center px-4 h-full cursor-pointer transition-all duration-200 text-white no-underline"
          data-discover="true"
          href="/contact"
        >
          Contact Us
        </a>
        <a
          class="font-mono text-xl flex items-center px-4 h-full cursor-pointer transition-all duration-200 text-white no-underline"
          data-discover="true"
          href="/sign-up"
        >
          Sign Up
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
